kore/src/Kore/Step.hs

```diff
@@ -118,6 +136,59 @@ transitionRule =
             & lift
         asum (pure <$> toList (Step.gatherResults results))
 
+{- | Transition rule for primitive strategies in 'Prim'.
+
+@transitionRule@ is intended to be partially applied and passed to
+'Strategy.runStrategy'.
+ -}
+transitionRule
+    ::  forall simplifier
+    .   MonadSimplify simplifier
+    =>  TransitionRule simplifier
+            (RewriteRule RewritingVariableName)
+            (ExecutionState (Pattern RewritingVariableName))
+transitionRule = transitionRuleWorker
+  where
+    transitionRuleWorker Begin (Remaining _) = empty
+    transitionRuleWorker Begin state = pure state
+
+    transitionRuleWorker Simplify (Rewritten patt) =
+        Rewritten <$> transitionSimplify patt
+    transitionRuleWorker Simplify (Remaining patt) =
+        Remaining <$> transitionSimplify patt
+
+    transitionRuleWorker (Rewrite rule) (Rewritten patt) =
+        transitionRewrite rule patt
```

Comment:
We should not rewrite states that have already been rewritten. Only one rewrite rule should apply in each `transitionRule` step.

Reply:
This would mean that we need another state, something like `Claimed` from the reachability `ClaimState`, right?

Reply:
I think `Remainder` serves that purpose here. We just should not apply rules to the `Rewritten` part.